src/rgw/rgw_admin.cc

```diff
@@ -1613,12 +1613,6 @@ static int commit_period(RGWRealm& realm, RGWPeriod& period,
     return ret;
   }
 
-  if (remote.empty() && url.empty()) {
-    // use the new master zone's connection
-    remote = master_zone;
-    cout << "Sending period to new master zone " << remote << std::endl;
-  }
-
```

Comment:
this isn't quite right - the `period` being committed here isn't necessarily the same as the period loaded by `RGWRados`, which is what `rest_master_conn` is based on. so if this `period commit` is changing which zone is the master, we'll send the commit to the wrong zone